src/components/Picker/BasePicker.tsx

```diff
@@ -0,0 +1,194 @@
+import lodashDefer from 'lodash/defer';
+import React, {ForwardedRef, forwardRef, ReactElement, ReactNode, RefObject, useContext, useEffect, useImperativeHandle, useRef, useState} from 'react';
+import {ScrollView, View} from 'react-native';
+import RNPickerSelect from 'react-native-picker-select';
+import FormHelpMessage from '@components/FormHelpMessage';
+import Icon from '@components/Icon';
+import * as Expensicons from '@components/Icon/Expensicons';
+import {ScrollContext} from '@components/ScrollViewWithContext';
+import Text from '@components/Text';
+import styles from '@styles/styles';
+import themeColors from '@styles/themes/default';
+import type {BasePickerHandle, BasePickerProps, PickerSize} from './types';
+
+const getDefaultPickerIcon = (iconSize: PickerSize): ReactElement => (
+    <Icon
+        src={Expensicons.DownArrow}
+        // eslint-disable-next-line react/jsx-props-no-spreading
+        {...(iconSize === 'small' ? {width: styles.pickerSmall().icon.width, height: styles.pickerSmall().icon.height} : {})}
+    />
+);
+
+function BasePicker<TPickerValue>(
+    {
+        items,
+        backgroundColor,
+        inputID,
+        value,
+        onInputChange,
+        label = '',
+        isDisabled = false,
+        errorText = '',
+        hintText = '',
+        containerStyles = [],
+        placeholder = {},
+        size = 'normal',
+        icon = getDefaultPickerIcon,
+        shouldFocusPicker = false,
+        onBlur = () => {},
+        additionalPickerEvents = () => {},
+    }: BasePickerProps<TPickerValue>,
+    ref: ForwardedRef<BasePickerHandle>,
+) {
+    const [isHighlighted, setIsHighlighted] = useState(false);
+
+    // reference to the root View
+    const root = useRef<View>(null);
+
+    // reference to @react-native-picker/picker
+    const picker = useRef<RNPickerSelect>(null);
+
+    // Windows will reuse the text color of the select for each one of the options
+    // so we might need to color accordingly so it doesn't blend with the background.
+    const pickerPlaceholder = Object.keys(placeholder).length ? {...placeholder, color: themeColors.pickerOptionsTextColor} : {};
+
+    useEffect(() => {
+        if (!!value || !items || items.length !== 1 || !onInputChange) {
+            return;
+        }
+
+        // When there is only 1 element in the selector, we do the user a favor and automatically select it for them
+        // so they don't have to spend extra time selecting the only possible value.
+        onInputChange(items[0].value, 0);
+
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [items]);
+
+    const context = useContext(ScrollContext);
```

Comment:
I think it would be good to move it to ScrollContext file and extract it to seperate function lik 
```suggestion
    const useScrollContext = () => useContext(ScrollContext);
```
so then we only need to import `useScrollContext` here

Reply:
Separate custom hook is what I was suggesting here, so I would go with similar approach as  [useLocalize](https://github.com/Expensify/App/blob/main/src/hooks/useLocalize.ts) 😄